beebop/app.py

```diff
@@ -15,16 +34,107 @@ def response_success(data):
     return response
 
 
+def response_failure(error):
+    response = {
+        "status": "failure",
+        "errors": [error],
+        "data": []
+    }
+    return response
+
+
+def check_connection(redis):
+    try:
+        redis.ping()
+        return True
+    except (redis_exceptions.ConnectionError, ConnectionRefusedError):
+        return jsonify(response_failure("Connection to redis failed"))
+
+
 @app.route('/version')
 def report_version():
     """
-    report version of beebop (and poppunk,ska in the future)
+    report version of beebop and poppunk (and ska in the future)
     wrapped in response object
     """
     vers = versions.get_version()
-    response = response_success(vers)
-    response_json = jsonify(response)
-    return response_json
+    return jsonify(response_success(vers))
+
+
+@app.route('/poppunk', methods=['POST'])
+@expects_json(schemas.sketches)
+def run_poppunk():
+    """
+    run poppunks assing_query()
+    input: multiple sketches in json format
+    output: clusters assigned to sketches
+    """
+    sketches = request.json['sketches'].items()
+    project_hash = request.json['projectHash']
+    return run_poppunk_internal(sketches, project_hash, storageLocation, redis)
+
+
+def run_poppunk_internal(sketches, project_hash, storageLocation, redis):
+    # create FS
+    fs = PoppunkFileStore(storageLocation)
+    # store json sketches in storage
+    hashes_list = []
+    for key, value in sketches:
+        hashes_list.append(key)
+        fs.input.put(key, value)
+    # check connection to redis
+    check = check_connection(redis)
+    if check:
+        # submit list of hashes to redis worker
+        q = Queue(connection=redis)
+        job = q.enqueue(assignClusters.get_clusters,
+                        hashes_list, project_hash, fs)
+        # save p-hash with job.id in redis server
+        redis.hset("beebop:hash:job", project_hash, job.id)
+        return job.id
+    else:
+        return check
```

Comment:
```suggestion
    check_connection(redis)
    # submit list of hashes to redis worker
    q = Queue(connection=redis)
    job = q.enqueue(assignClusters.get_clusters,
                    hashes_list, project_hash, fs)
    # save p-hash with job.id in redis server
    redis.hset("beebop:hash:job", project_hash, job.id)
    return job.id
```

And elsewhere - no need to handle the logic here where you return `False` from a function that never returns that. Instead, just assume that if you've not thrown in `check_connection` that you can sail along happily. You can make this more explicit by renaming `check_connection` to `assert_connection_valid` but it's fine as is. Same change to other endpoints too